go/common/async/async.go

```diff
@@ -0,0 +1,53 @@
+package async
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type (
+	ScheduledFunc func()
+)
+
+// ScheduleInterrupt runs the function after the delay and can be interrupted
+func ScheduleInterrupt(delay time.Duration, interrupt *int32, fun ScheduledFunc) {
+	ticker := time.NewTicker(delay)
+
+	go func() {
+		<-ticker.C
+		if atomic.LoadInt32(interrupt) == 1 {
+			return
+		}
+
+		fun()
+		ticker.Stop()
+	}()
+}
```

Comment:
I think you can delete this method because the interrupt it checked on the `ScheduledFunc`.